<?xml version="1.0"?>
<launch>
    <!-- support multi sensor in one master -->
    <arg name="sensor_name" default=""/>   <!-- support multi robot -->
    <arg name="port" default="sensor_imu"/>
    <arg name="device_frame"       default="imu_module_frame" />
    <!-- frame -->
    <arg name="imu_frame"       default="imu_frame" /> 
    <arg name="mag_frame"       default="mag_frame" />    
    <!-- pub topic -->
    <arg name="imu_topic"       default="imu" />
    <arg name="mag_topic"       default="mag" />
    <!-- topic freq -->
    <arg name="imu_freq"       default="200" />
    <arg name="mag_freq"       default="70" />
    <arg name="gravity"       default="true" />
    <!--ros namespace name can NOT be empty,so need evaluate use multi robot or not-->
    <group if="$(eval sensor_name == '')">
        <node name="sensor_imu"   pkg="sensor_imu"  type="sensor_imu.py" output="screen">
            <param name="baudrate"   value="115200"/>  
            <param name="port"  value='/dev/$(arg port)'/>
            <param name="imu_id"    value="$(arg imu_frame)"/>      <!-- imu link name -->
            <param name="imu_topic" value="$(arg imu_topic)"/>      <!-- topic name -->
            <param name="imu_freq" value="$(arg imu_freq)"/>        <!-- topic freq -->

            <param name="mag_id"    value="$(arg mag_frame)"/>      <!-- link name -->
            <param name="mag_topic" value="$(arg mag_topic)"/>      <!-- topic name -->
            <param name="mag_freq" value="$(arg mag_freq)"/>        <!-- topic freq -->

            <param name="gravity" value="$(arg gravity)"/>          <!-- use gravity -->
        </node>
        <node pkg="tf" type="static_transform_publisher" name="module_to_imu"
            args="0.007 0.0 0.0 0.0 0.0 0.0 $(arg device_frame) $(arg imu_frame) 20">
        </node>
        <node pkg="tf" type="static_transform_publisher" name="module_to_mag"
            args="0.007 0.0 0.0 0.0 0.0 0.0 $(arg device_frame) $(arg mag_frame) 20">
        </node>
    </group>

    <group unless="$(eval sensor_name == '')">
        <group ns="$(arg sensor_name)">
            <node name="sensor_imu"   pkg="sensor_imu"  type="sensor_imu.py" output="screen">
                <param name="baudrate"   value="115200"/>  
                <param name="port"  value='/dev/$(arg port)'/>
                <param name="imu_id"    value="$(arg sensor_name)/$(arg imu_frame)"/>      <!-- imu link name -->
                <param name="imu_topic" value="$(arg imu_topic)"/>      <!-- topic name -->
                <param name="imu_freq" value="$(arg imu_freq)"/>        <!-- topic freq -->

                <param name="mag_id"    value="$(arg sensor_name)/$(arg mag_frame)"/>      <!-- link name -->
                <param name="mag_topic" value="$(arg mag_topic)"/>      <!-- topic name -->
                <param name="mag_freq" value="$(arg mag_freq)"/>        <!-- topic freq -->
                <param name="gravity" value="$(arg gravity)"/>          <!-- use gravity -->
            </node>
            <node pkg="tf" type="static_transform_publisher" name="module_to_imu"
                args="0.007 0.0 0.0 0.0 0.0 0.0 $(arg sensor_name)/$(arg device_frame) $(arg sensor_name)/$(arg imu_frame) 20">
            </node>
            <node pkg="tf" type="static_transform_publisher" name="module_to_mag"
                args="0.007 0.0 0.0 0.0 0.0 0.0 $(arg sensor_name)/$(arg device_frame) $(arg sensor_name)/$(arg mag_frame) 20">
            </node>
        </group>
    </group>

</launch>
